src/scheduler/fifo.c

```diff
@@ -2327,3 +2350,357 @@ next_job(status *policy, server_info *sinfo, int flag)
+ * @param[in] dest - pointer to source string
+ *
+ * @retval
+ * @return 0 - Failure
+ * @return  1 - Success
+ *
+ * @par Side Effects:
+ *	None
+ *
+ *
+ */
+static int
+copy_attr_value(char **dest, char *src)
+{
+	int len = 0;
+	int ret = 0;
+
+	if (*dest != NULL)
+		free(*dest);
+
+	if (src !=NULL) {
+		len = strlen(src);
+		*dest = (char*)malloc(len + 1);
+		if (*dest == NULL) {
+			log_err(errno, __func__, MEM_ERR_MSG);
+			return ret;
+		}
+		strncpy(*dest, src, len);
+		(*dest)[len] = '\0';
+		ret = 1;
+	}
+	return ret;
+}
+
+/**
+ * @brief
+ *	Helper function used to copy the attribute values from batch_status to the corresponding
+ *	scheduler global variables which hold its priv_dir, log_dir and partitions
+ *
+ * @param[in] status - populated batch_status after stating this scheduler from server
+ *
+ * @retval
+ * @return 0 - Failure
+ * @return  1 - Success
+ *
+ * @par Side Effects:
+ *	None
+ *
+ *
+ */
+static int
+sched_settings_frm_svr(struct batch_status *status)
+{
+	struct attrl *attr;
+	char *tmp_priv_dir = NULL;
+	char *tmp_log_dir = NULL;
+	char *tmp_partitions = NULL;
+	struct	attropl	*attribs;
+	static char *log_dir = NULL;
+	char *tmp_comment = NULL;
+	int clear_comment = 0;
+	extern char *partitions;
+
+	attr = status->attribs;
+
+	 /* resetting the following before fetching from batch_status. */
+	while (attr != NULL) {
+		if (attr->name != NULL && attr->value != NULL) {
+			if (!strcmp(attr->name, ATTR_sched_priv)) {
+				if (copy_attr_value(&tmp_priv_dir, attr->value) == 0)
+					return 0;
+			} else if (!strcmp(attr->name, ATTR_sched_log)) {
+				if (copy_attr_value(&tmp_log_dir, attr->value) == 0)
+					return 0;
+			} else if (!strcmp(attr->name, ATTR_partition)) {
+				if (copy_attr_value(&tmp_partitions, attr->value) == 0)
+					return 0;
+			} else if (!strcmp(attr->name, ATTR_comment)) {
+				if (copy_attr_value(&tmp_comment, attr->value) == 0)
+					return 0;
+			}
+		}
+		attr = attr->next;
+	}
+
+	if (!dflt_sched) {
+		int err;
+		int priv_dir_update_fail = 0;
+		struct attropl *patt;
+		char comment[MAX_LOG_SIZE] = {0};
+		static char *priv_dir = NULL;
+		if ((log_dir != NULL) && strcmp(log_dir, tmp_log_dir) != 0) {
+			(void)snprintf(path_log,  sizeof(path_log), tmp_log_dir);
+			log_close(1);
+			if (log_open(logfile, path_log) == -1) {
+				/* update the sched comment attribute with the reason for failure */
+				attribs = calloc(2, sizeof(struct attropl));
+				if (attribs == NULL) {
+					snprintf(log_buffer, sizeof(log_buffer), "can't update scheduler attribs, calloc failed");
+					log_err(errno, __func__, log_buffer);
+					free(tmp_log_dir);
+					free(tmp_priv_dir);
+					free(tmp_partitions);
+				}
+
+				strncpy(comment, "Unable to change the sched_logs directory", MAX_LOG_SIZE -1);
+				patt = attribs;
+				patt->name = ATTR_comment;
+				patt->value = comment;
+				patt->next = patt + 1;
+				patt++;
+				patt->name = ATTR_scheduling;
+				patt->value = "0";
```

Comment:
This is done as per the suggestion of Bhroam. This will ensure that jobs will not be scheduled by a corresponding scheduler if its configuration set by the admin is invalid.

Reply:
I guess we have two options here.  We could set scheduling to false or we can just mark the scheduler down.  I'm fine with either.

Reply:
Now that I think about it more, I like setting scheduling to false as well.  Having a scheduler as down has a specific meaning.  It means the server can not connect to the scheduler.  It would muddy the waters if we had it mean multiple things.